pageserver/src/tenant.rs

```diff
@@ -2324,19 +2150,25 @@ impl Tenant {
     /// The returned Timeline is in Loading state. The caller is responsible for
     /// initializing any on-disk state, and for inserting the Timeline to the 'timelines'
     /// map.
+    ///
+    /// `validate_ancestor == false` is used when a timeline is created for deletion
+    /// and we might not have the ancestor present anymore which is fine for to be
+    /// deleted timelines.
     fn create_timeline_struct(
         &self,
         new_timeline_id: TimelineId,
         new_metadata: &TimelineMetadata,
         ancestor: Option<Arc<Timeline>>,
         remote_client: Option<RemoteTimelineClient>,
         init_order: Option<&InitializationOrder>,
+        validate_ancestor: bool,
     ) -> anyhow::Result<Arc<Timeline>> {
```

Comment:
At the minimum, can we turn `validate_ancestor` into an enum? Also, it can be more expressive.

```
`enum CreateTimelineStructCause {
  Load,
  Delete,
}
```

and then make the decision to validate ancestor based on that.

Reply:
Will do

Reply:
Done